bridge/rusty_webf_sys/src/native_value.rs

```diff
@@ -0,0 +1,225 @@
+
+    let shared_string_ptr = shared_string_ptr as *mut SharedNativeString;
+    unsafe {
+      shared_string_ptr.write(shared_string);
+    }
+
+    let mut value = Self::new();
+    value.tag = NativeTag::TagString as i32;
+    value.u.ptr = shared_string_ptr as *mut c_void;
+    value.uint32 = len as u32;
+    value
+  }
+
+  pub fn to_string(&self) -> String {
+    let ptr = unsafe {
+      self.u.ptr as *mut SharedNativeString
+    };
+    let string_struct = unsafe { ptr.read() };
+    let slice = unsafe { std::slice::from_raw_parts(string_struct.string_, string_struct.length_.try_into().unwrap()) };
+    String::from_utf16_lossy(slice)
+  }
+
+  pub fn new_null() -> Self {
+    let mut value = Self::new();
+    value.tag = NativeTag::TagNull as i32;
+    value.u.int64 = 0;
+    value.uint32 = 0;
+    value
+  }
+
+  pub fn is_null(&self) -> bool {
+    self.tag == NativeTag::TagNull as i32
+  }
+
+  pub fn new_float64(val: f64) -> Self {
+    let mut value = Self::new();
+    value.tag = NativeTag::TagFloat64 as i32;
+    value.u.float64 = val;
+    value.uint32 = 0;
+    value
+  }
+
+  pub fn to_float64(&self) -> f64 {
+    unsafe {
+      self.u.float64
+    }
+  }
+
+  pub fn new_bool(val: bool) -> Self {
+    let mut value = Self::new();
+    value.tag = NativeTag::TagBool as i32;
+    value.u.int64 = if val { 1 } else { 0 };
+    value.uint32 = 0;
+    value
+  }
+
+  pub fn to_bool(&self) -> bool {
+    unsafe {
+      self.u.int64 != 0
+    }
+  }
+
+  pub fn new_int64(val: i64) -> Self {
+    let mut value = Self::new();
+    value.tag = NativeTag::TagInt as i32;
+    value.u.int64 = val;
+    value.uint32 = 0;
+    value
+  }
+
+  pub fn to_int64(&self) -> i64 {
+    unsafe {
+      self.u.int64
+    }
+  }
+
+  pub fn new_list(values: Vec<NativeValue>) -> Self {
+    let size = values.len();
+    let array_size = size * mem::size_of::<NativeValue>();
+
+    #[cfg(target_os = "windows")]
+    let array_ptr = unsafe { CoTaskMemAlloc(array_size) };
+
+    #[cfg(not(target_os = "windows"))]
+    let array_ptr = unsafe { libc::malloc(array_size) };
+
+    let array_ptr = array_ptr as *mut NativeValue;
+
+    for (i, val) in values.iter().enumerate() {
+      unsafe {
+        array_ptr.add(i).write(val.clone());
+      }
+    }
+
+    let mut value = Self::new();
+    value.tag = NativeTag::TagList as i32;
+    value.u.ptr = array_ptr as *mut c_void;
+    value.uint32 = size as u32;
+    value
+  }
+
+  pub fn to_list(&self) -> Vec<NativeValue> {
+    let mut values = Vec::new();
+    let ptr = unsafe {
+      self.u.ptr as *mut NativeValue
+    };
+    for i in 0..self.uint32 {
+      let offset = i.try_into().unwrap();
+      let val = unsafe { ptr.add(offset).read() };
+      values.push(val);
+    }
+    values
+  }
+}
+
+impl Drop for NativeValue {
+  fn drop(&mut self) {
+    if self.tag == NativeTag::TagString as i32 {
+      println!("Drop NativeValue string: {}", self.to_string());
```

Comment:
Missing impl for release string heap in rust

Reply:
updated.